lib/Mitigation.pm

```diff
@@ -0,0 +1,352 @@
+#    sysfs_name             => "l1tf",
+#    sysfs                  => {
+#        "full"         => "Mitigation: PTE Inversion; VMX: cache flushes, SMT disabled",
+#        "full,force"   => "Mitigation: PTE Inversion; VMX: cache flushes, SMT disabled",
+#        "flush"        => "Mitigation: PTE Inversion; VMX: conditional cache flushes, SMT vulnerable",
+#        "flush,nosmt"  => "Mitigation: PTE Inversion; VMX: conditional cache flushes, SMT disabled",
+#        "flush,nowarn" => "Mitigation: PTE Inversion; VMX: conditional cache flushes, SMT vulnerable",
+#        "off"          => "Mitigation: PTE Inversion; VMX: vulnerable",
+#        "default"      => "Mitigation: PTE Inversion; VMX: conditional cache flushes, SMT vulnerable",
+#    },
+#    cmdline => [
+#        "full",
+#        "full,force",
+#        "flush",
+#        "flush,nosmt",
+#        "flush,nowarn",
+#        "off",
+#    ],
+#  );
+#
+#  "name", the name of test module or mitigations name.
+#  "CPUID", the bit that should be check via cpuid instruction. follow Intel manual.
+#  "IA32_ARCH_CAPABILITIES", the bit that means this mitigations might be fixed by Hardware.
+#  "parameter", the name of parameter on kernel cmdline, as switch to enable or disable this mitigation in system.
+#  "cpuflags", the flag[s] name should be appear at /proc/cpuinfo or lscpu.
+#  "sysfs_name", the name of entry in sysfs: /sys/devices/system/cpu/vulnerabilities/*,
+#  "sysfs", a hash with  {"name" => "string"}
+#  	Means, when "name" be used as a kernel parameter to value, the "string" should be the context of "sysfs_name".
+#  "cmdline", an array that store what kernel parameter will be tested.
+#
+#TODO:
+#In the future, there will be a dmesg item, it should seems like "sysfs", that could be used for check the some messages in dmesg log.
+#
+## Maintainer: James Wang <jnwang@suse.com>
+#
+#
+package Mitigation;
+
+use strict;
+use warnings;
+
+use base "Exporter";
+use base "opensusebasetest";
+use Exporter;
+
+use testapi;
+use utils;
+
+use Utils::Backends 'use_ssh_serial_console';
+use bootloader_setup qw(grub_mkconfig change_grub_config add_grub_cmdline_settings remove_grub_cmdline_settings grep_grub_settings set_framebuffer_resolution set_extrabootparams_grub_conf);
+use ipmi_backend_utils;
+use power_action_utils 'power_action';
+
+
+=head2 reboot_and_wait
+
+	reboot_and_wait([timeout => $timeout]);
+
+To reboot and waiting system back and login it.
+This could support IPMI and QEMU backend.
+C<$timeout> in seconds.
+=cut
+sub reboot_and_wait {
+    my ($self, $timeout) = @_;
+    power_action('reboot', textmode => 1, keepconsole => 1);
+    if (check_var('BACKEND', 'ipmi')) {
+        switch_from_ssh_to_sol_console(reset_console_flag => 'on');
+        check_screen('login_screen', $timeout);
+        use_ssh_serial_console;
+    }
+    else {
+        $self->wait_boot(textmode => 1, ready_time => 300,, in_grub => 1);
+        select_console 'root-console';
+    }
+}
```

Comment:
I don't see why we should need another "reboot_and_wait" function. We have enough of these and you already use them. Please don't add any "if (backend=foo) { … }" code. If there is something missing in the utility functions then add your suggestions there.

Reply:
But if this code only be used in my testing, I also need to modify the common code?

Reply:
Hi @okurz ,I think I need these lines, for now, I use some of the functions of "reboot_and_wait" in a different way with others.